Update CONTRIBUTING.md (#1)
litestar-org/.github: CONTRIBUTING.md

1. Install [`poetry`](https://python-poetry.org/)
2. Run `poetry install` to create a [`virtual environment`](https://docs.python.org/3/tutorial/venv.html) and install
   the dependencies

> 3. If you're working on the documentation and need to build it locally, install the extra dependencies with `poetry install --with dev`

Reviewer: ```suggestion
3. If you're working on the documentation and need to build it locally, install the extra dependencies with `poetry install`
```

> 3. If you're working on the documentation and need to build it locally, install the extra dependencies with `poetry install`
> 4. Install [`pre-commit`](https://pre-commit.com/)
> 5. Run `pre-commit install` to install pre-commit hooks

Reviewer: ```suggestion
3. Install [`pre-commit`](https://pre-commit.com/)
4. Run `pre-commit install` to install pre-commit hooks
```

- All functions, methods, classes and attributes should be documented with a docstring. We use the
  [`Google docstring style`](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). If you come
  across a function or method that doesn't conform to this standard, please update it as you go

### Writing and running tests

> Tests should be contained within the `tests` directory, and follow the same directory structure as the `starlite` module.

Reviewer: ```suggestion
Tests should be contained within the `tests` directory, and follow the same directory structure as the main package.
```

If you are adding a test case, it should be located within the correct submodule of `tests`. E.g. tests for
`starlite/utils/sync.py` reside in `tests/utils/test_sync.py`.

If the repository you are working on includes a `Makefile` you have access to includes several commands for running tests:

- `make test` to run tests located in `tests`
- `make test-examples` to run tests located in `docs/examples/tests`
- `make test-all` to run all tests
- `make coverage` to run tests with coverage and generate a html report

> If not, you are able to manually run `poetry run pytest` or `pytest tests/`

Reviewer: I think this section can be removed. The `Makefile` is very specific to the Starlite repo
